addons/sms/wizard/sms_composer.py

```diff
@@ -86,7 +52,39 @@ def default_get(self, fields):
     sanitized_numbers = fields.Char('Sanitized Number', compute='_compute_sanitized_numbers', compute_sudo=False)
     # content
     template_id = fields.Many2one('sms.template', string='Use Template', domain="[('model', '=', res_model)]")
-    body = fields.Text('Message', required=True)
+    body = fields.Text('Message', required=True, store=True, compute='_compute_body', readonly=False)
+
+    @api.depends('active_domain', 'res_id', 'res_ids')
+    def _compute_composition_mode(self):
```

Comment:
Well here I think you went a bit too far at default_get removal :) . We should distinguish

  * real default values computation
  * default values computation that simulates onchange / dependencies

I would therefore keep code related to res_model / res_id / res_ids computation in default_get . Indeed this is just some initialization. Those value will not change during the composer lifetime, we just have some boilerplate to ease future computation.

I think the only real field that should be a compute is body, actually. Maybe composition_mode, in which case you have to explicitely tell it depends on the context ( api.depends_context ) .

It should simplify this PR quite a lot :) .